modules/apps/wiki/wiki-service/src/main/java/com/liferay/wiki/service/impl/WikiPageLocalServiceImpl.java

```diff
@@ -3431,6 +3477,36 @@ private void _validateExternalReferenceCode(
 		}
 	}
 
+	private void _validateHierarchyCycle(
+			long nodeId, String title, String parentTitle)
+		throws PortalException {
+
+		if (Validator.isNull(title) || Validator.isNull(parentTitle)) {
+			return;
+		}
+
+		title = title.trim();
+
+		while (Validator.isNotNull(parentTitle)) {
+			parentTitle = parentTitle.trim();
+
+			if (StringUtil.equalsIgnoreCase(title, parentTitle)) {
+				throw new PortalException(
+					"Unable to update wiki page " + title +
+						" because a hierarchy cycle was detected");
+			}
+
+			WikiPage parentPage = fetchPage(nodeId, parentTitle);
+
+			if (parentPage != null) {
+				parentTitle = parentPage.getParentTitle();
+			}
+			else {
+				return;
```

Comment:
Instead of the explicit return, can we set `parentTitle` to `null` so that the loop ends naturally?

Reply:
Sure!!